web_src/js/components/file_tree.ts

```diff
@@ -0,0 +1,90 @@
+export type FileStatus = 'added' | 'modified' | 'deleted' | 'renamed' | 'copied' | 'typechange';
+
+export type File = {
+  Name: string;
+  NameHash: string;
+  Status: FileStatus;
+  IsViewed: boolean;
+  IsSubmodule: boolean;
+}
+
+type DirItem = {
+    isFile: false;
+    name: string;
+    path: string;
+
+    // we disable eslint here because we have a recursive type definition
+    // eslint-disable-next-line no-use-before-define
```

Comment:
This is likely a configuration issue, we should be using `@typescript-eslint/no-use-before-define` instead which should not raise this issue. I will check later.

https://typescript-eslint.io/rules/no-use-before-define/

Reply:
Will be fixed with https://github.com/go-gitea/gitea/pull/33715 after which this rule disablement can be removed.